include/cocaine/rpc/asio/decoder.hpp

```diff
@@ -49,8 +51,20 @@ struct decoded_message_t {
         return object.via.array.ptr[2];
     }
 
+    boost::optional<header_t>
+    get_header(const header_key_t& key) const {
```

Comment:
Why not? It doesn't change anything in fact, as header_t can not modify data it points to and semantic is close to "return a copy"